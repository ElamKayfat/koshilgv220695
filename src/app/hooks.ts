import { useEffect } from 'react';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import { setUserListData } from '../features/UserList/UserListSlice';
import { mockUserList } from './mockData';
import type { RootState, AppDispatch } from './store';

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export const useFetchUserListAndSave = () => {
  const dispatch = useDispatch()
  
  useEffect(() => {
    setTimeout(() => {
      dispatch(setUserListData({ userListData: mockUserList }))
    }, 200)
  }, [])
}
