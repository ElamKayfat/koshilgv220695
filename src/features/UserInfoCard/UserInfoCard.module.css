.root {
  max-width: 400px;
  max-height: 500px;
  margin: auto;
  margin-top: 50px;
  border: 1px solid lightgray;
  padding: 30px 15px 15px 15px;
  position: relative;
}

.userAboutContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.photo {
  width: 100px;
  height: 100px;
  border-radius: 50%;
}

.name {
  font-size: 16px;
  font-weight: 600;
  margin-top: 5px;
}

.position {
  font-size: 14px;
  color: darkgray;
  margin-top: 3px;
}

.divider {
  height: 1px;
  background: lightgray;
  margin: 20px 0;
}

.sendButton {
  margin: auto;
  border: none;
  border-radius: 6px;
  width: 100%;
  color: white;
  background: rgb(15, 122, 158);
  padding: 10px;
  font-size: 14px;
  margin-top: 20px;
}

.sendButton:hover {
  opacity: 0.8;
}

.privateInfoName {
  font-size: 16px;
  margin-right: 20px;
  font-weight: 600;
  display: block;
  width: 50px;
}

.userPrivateInfoItem {
  display: flex;
  margin-bottom: 5px;
}

.blue {
  color: rgb(44, 40, 241);
}

.closeButton {
  font-weight: 600;
  font-size: 22px;
  position: absolute;
  width: 20px;
  height: 20px;
  top: 10px;
  right: 10px;
  cursor: pointer;
}

.closeButton:hover {
  opacity: 0.8;
}