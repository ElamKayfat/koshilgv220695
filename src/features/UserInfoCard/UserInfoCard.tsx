import { UserDataType } from "../UserList/UserListSlice";
import { capitalizeFirstLetter } from "../../app/utils";
import styles from "./UserInfoCard.module.css";
import { FC } from "react";

type UserInfoCardProps = {
  userData: UserDataType;
  onClose: () => void;
};

function UserInfoCard(props: UserInfoCardProps) {
  return (
    <div className={styles.root}>
      <CloseButton onClose={props.onClose} />

      <MainInfoSection userData={props.userData} />

      <Divider />

      <PrivateInfoSection userData={props.userData} />

      <SendButton />
    </div>
  );
}

export default UserInfoCard;

const CloseButton: FC<{ onClose: () => void }> = ({ onClose }) => (
  <div
    className={styles.closeButton}
    onClick={onClose}
    data-testid="close-button"
  >
    x
  </div>
);

const MainInfoSection: FC<{ userData: UserDataType }> = ({ userData }) => (
  <div className={styles.userAboutContainer}>
    <img className={styles.photo} src={userData.photo} alt={userData.name} />
    <span className={styles.name}>{userData.name}</span>
    <span className={styles.position}>{userData.position}</span>
  </div>
);

const Divider = () => <div className={styles.divider} />;

const PrivateInfoSection: FC<{ userData: UserDataType }> = ({ userData }) => (
  <div className={styles.userPrivateInfoContainer}>
    {["phone", "url", "email"].map((itemName: string) => (
      <div className={styles.userPrivateInfoItem} key={itemName}>
        <span className={styles.privateInfoName}>
          {capitalizeFirstLetter(itemName)}
        </span>
        <span
          className={
            itemName !== "phone"
              ? `${styles.privateInfoValue + styles.blue} ${styles.blue}`
              : styles.privateInfoValue
          }
        >
          {userData[itemName as keyof UserDataType] ?? "No have a data"}
        </span>
      </div>
    ))}
  </div>
);

const SendButton = () => (
  <button className={styles.sendButton}>Send message</button>
);
