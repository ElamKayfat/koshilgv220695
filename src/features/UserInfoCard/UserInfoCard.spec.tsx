import { fireEvent, getByTestId, getByText, render } from '@testing-library/react';
import { mockUserList } from '../../app/mockData';
import UserInfoCard from './UserInfoCard';

test('Make snapshot. Render with data', () => {
  const { container } = render(<UserInfoCard userData={mockUserList[1]} onClose={() => console.log("close")} />);

  expect(container).toMatchSnapshot();
});

test('Make snapshot. Click on close button', () => {
  const mockOnClose = jest.fn()
  const { container } = render(<UserInfoCard userData={mockUserList[1]} onClose={mockOnClose} />);
  
  fireEvent(
    getByTestId(container, 'close-button'),
    new MouseEvent('click', {
      bubbles: true,
      cancelable: true,
    }),
  )
  
  expect(mockOnClose.mock.calls).toHaveLength(1);;
});