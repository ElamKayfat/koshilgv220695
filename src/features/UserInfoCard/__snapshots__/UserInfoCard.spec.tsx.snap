// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make snapshot. Render with data 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="closeButton"
      data-testid="close-button"
    >
      x
    </div>
    <div
      class="userAboutContainer"
    >
      <img
        alt="Brazil Izair"
        class="photo"
        src="2.jpg"
      />
      <span
        class="name"
      >
        Brazil Izair
      </span>
      <span
        class="position"
      >
        Managing Director
      </span>
    </div>
    <div
      class="divider"
    />
    <div
      class="userPrivateInfoContainer"
    >
      <div
        class="userPrivateInfoItem"
      >
        <span
          class="privateInfoName"
        >
          Phone
        </span>
        <span
          class="privateInfoValue"
        >
          902-782-3286
        </span>
      </div>
      <div
        class="userPrivateInfoItem"
      >
        <span
          class="privateInfoName"
        >
          Url
        </span>
        <span
          class="privateInfoValueblue blue"
        >
          https://example.com
        </span>
      </div>
      <div
        class="userPrivateInfoItem"
      >
        <span
          class="privateInfoName"
        >
          Email
        </span>
        <span
          class="privateInfoValueblue blue"
        >
          brazil.izair@fallinhay.com
        </span>
      </div>
    </div>
    <button
      class="sendButton"
    >
      Send message
    </button>
  </div>
</div>
`;
